src/test/shared/defaultAwsContext.test.ts

```diff
@@ -169,55 +169,39 @@ describe('DefaultAwsContext', () => {
         await testContext.addExplorerRegion(testRegion1Value)
-    })
-
-    it('updates config on multiple region change', async () => {
-        class TestConfiguration extends ContextTestsSettingsConfigurationBase {
-            public async writeSetting<T>(settingKey: string, value: T, target: ConfigurationTarget): Promise<void> {
-                assert.strictEqual(settingKey, regionSettingKey)
-                assert(value instanceof Array)
-                const values = (value as any) as string[]
-                assert.strictEqual(values[0], testRegion1Value)
-                assert.strictEqual(values[1], testRegion2Value)
-                assert.strictEqual(target, ConfigurationTarget.Global)
-            }
-        }
 
-        const testContext = new DefaultAwsContext(new TestConfiguration(), new FakeExtensionContext())
-        await testContext.addExplorerRegion(testRegion1Value, testRegion2Value)
+        const persistedRegions = extensionContext.globalState.get<string[]>(regionSettingKey)
+        assert.ok(persistedRegions, 'Expected region data to be stored in globalState')
+        assert.strictEqual(persistedRegions!.length, 1)
+        assert.strictEqual(persistedRegions![0], testRegion1Value)
     })
 
-    it('updates on region removal', async () => {
-        class TestConfiguration extends ContextTestsSettingsConfigurationBase {
-            public readSetting<T>(settingKey: string, defaultValue?: T): T | undefined {
-                if (settingKey === regionSettingKey) {
-                    return ([testRegion1Value, testRegion2Value] as any) as T
-                }
+    it('updates globalState on multiple region change', async () => {
+        const extensionContext = new FakeExtensionContext()
+        const testContext = new DefaultAwsContext(new ContextTestsSettingsConfigurationBase(), extensionContext)
+        await testContext.addExplorerRegion(testRegion1Value, testRegion2Value)
 
-                return super.readSetting<T>(settingKey, defaultValue)
-            }
-            public async writeSetting<T>(settingKey: string, value: T, target: ConfigurationTarget): Promise<void> {
-                assert.strictEqual(settingKey, regionSettingKey)
-                assert.deepStrictEqual(value, [testRegion1Value])
-                assert.strictEqual(target, ConfigurationTarget.Global)
-            }
-        }
+        const persistedRegions = extensionContext.globalState.get<string[]>(regionSettingKey)
+        assert.ok(persistedRegions, 'Expected region data to be stored in globalState')
+        assert.strictEqual(persistedRegions!.length, 2)
+        assert.strictEqual(persistedRegions![0], testRegion1Value)
+        assert.strictEqual(persistedRegions![1], testRegion2Value)
+    })
 
-        const testContext = new DefaultAwsContext(new TestConfiguration(), new FakeExtensionContext())
+    it('updates globalState on region removal', async () => {
```

Comment:
Could you do the following:
* Add a validation to ensure that both regions are available before removing one
* Remove testRegion1Value instead of testRegion2Value so we can see if the array adjusts successfully

Reply:
> Add a validation to ensure that both regions are available before removing one

It is unnecessary - the two tests above this one verify that

> Remove testRegion1Value instead of testRegion2Value so we can see if the array adjusts successfully

I'll do you one better by adding 3 regions, and removing the middle region.

Reply:
Oh snaaaaaaaaaaaap